### PR TITLE
feat: create MetaMaskSDKConnector using @metamask/sdk

### PR DESCRIPTION
**Pull request by abretonc7s**
## Description

Add MetaMask SDK integration to allow MetaMask mobile wallet users to connect.

## Additional Information

- [ x ] I read the [contributing docs](/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)
- switch between browser extension and mobile wallet
- Kept metamask connector for backward compatibility and staged user migration

Your ENS/address: 0x5Cba62cA5F5C47d39943f38b2047B90bFfA41cd9

## Demo
wagmi_demo

### CODE REVIEW

**Comment by abretonc7s** [comment]
@SocketSecurity ignore @metamask/sdk

File: packages/connectors/package.json [hunk 1]
```diff
@@ -23,12 +23,13 @@
   "dependencies": {
     "@coinbase/wallet-sdk": "^3.6.6",
     "@ledgerhq/connect-kit-loader": "^1.1.0",
+    "@metamask/sdk": "^0.5.6",
```
Comment (tmm) [inline packages/connectors/package.json:26]: Would be great if `package.json#sideEffects` was added to `@metamask/sdk`.

Without `"sideEffects"` defined (e.g. `"sideEffects": false`) build systems aren't able to properly eliminate dead code (e.g. Next.js) and could still load these packages even if they are not used.

More info: https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Comment (abretonc7s) [reply packages/connectors/package.json:26]: done with sdk 0.6.0

**Comment by tmm** [comment]
Appreciate the effort that went into this! We've been busy with other work so haven't had time to support new connectors. We are reworking how connectors are setup (directly in the wagmi repo instead of references, new API, etc.) so once that is done this can be created again. Stayed tuned to the [wagmi discussion board](https://github.com/wagmi-dev/wagmi/discussions) for more info.

In the meantime, I would suggest releasing this as it's own package so folks can install and use directly from there.

**Pull request closed without merging**